### PR TITLE
[FIX] product: cannot change company when there are only exist one co…

### PR DESCRIPTION
**Pull request by paimonchan**
Impacted versions: 14.0

Steps to reproduce:

- have one company (company A)
- create new product without company
- create transaction from sale/purchase until the transaction is completed on company A
- create another company (company B) ==> don't do any transaction in this company
- set the product company to company A
- will get error message This product's company cannot be changed as long as there are quantities of it belonging to another company.

Current behavior:

- verification for changing product company was included for quant in vendor location and customer location

Expected behavior:

- verification for changing product company should be exclude vendor location and customer location, because this virtual location doesn't have a company

Video/Screenshot link (optional):

Support ticket number submitted via odoo.com/help (optional):

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by Whenrow** [comment]
@paimonchan Hello, thank you for the fix. However, I would suggest not to test the location usage but the company_id value. Quants with company_id falsy should be ignored

**Comment by paimonchan** [comment]
@Whenrow Roger, already updated to ignore quant from falsy company_id

File: addons/stock/models/product.py [hunk 1]
```diff
@@ -766,7 +766,7 @@ def write(self, vals):
             if products_changing_company:
                 # Forbid changing a product's company when quant(s) exist in another company.
                 quant = self.env['stock.quant'].sudo().search([
-                    ('location_id.usage', 'not in', ['customer', 'supplier']),
+                    ('company_id', '!=', False),
                     ('product_id', 'in', products_changing_company.product_variant_ids.ids),
                     ('company_id', '!=', vals['company_id']),
```
Comment (Whenrow) [inline addons/stock/models/product.py:771]: I was thinking about adding the filter in the existing one about company_id 
`('company_id', 'not in', [vals['company_id'], False])`,

Sorry if that was not clear

Comment (paimonchan) [reply addons/stock/models/product.py]: got it, done updated

**Comment by Whenrow** [comment]
@paimonchan Ty ;)
robodoo r+

**Comment by Whenrow** [comment]
robodoo rebase-ff

**Comment by paimonchan** [comment]
> @paimonchan I cannot merge this PR right now. Could you squash your commits (1 for the CLA and 1 for the code in stock module) ? I think this will help. Thank you

can i make it only 1 commit? i think the CLA already mixed with the code (in the first commit)

**Comment by Whenrow** [comment]
@paimonchan I prefer having the CLA commit alone (with the tag [CLA]). 
By 'squashing' I mean rewriting your branch history to only have 2 commits in the PR

**Comment by Whenrow** [comment]
robodoo r+

**Pull request closed without merging**